File: 1005a/main.py
def I():
    return input()


def II():
    return int(input())


def MII():
    return map(int, input().split())


def LI():
    return list(input().split())


def LII():
    return list(map(int, input().split()))

a = II()
L = LII()
L.append(1)
ans = []
tmp = 1
for i in range(1, a+1):
    if L[i] <= L[i-1]:
        ans.append(L[i-1])
print(len(ans))
print(*ans, sep = ' ')

File: 1005a/main.go
package main

import "fmt"

func main() {
	var a int
	fmt.Scan(&a)
	L := make([]int, a)
	for i := 0; i < a; i++ {
		fmt.Scan(&L[i])
	}
	L = append(L, 1)
	var ans []int
	//fmt.Println(L)
	for i := 1; i <= a; i++ {
		if L[i] <= L[i-1] {
			ans = append(ans, L[i-1])
		}
	}
	fmt.Println(len(ans))
	for _, v := range ans {
		fmt.Print(v, " ")
	}
	fmt.Println()
}
